File: Samples/azure-storage/Azure.CSharp/Models/Usage.cs

namespace Petstore.Models
{
    using System.Linq;

    /// <summary>
    /// Describes Storage Resource Usage.
    /// </summary>
    public partial class Usage
    {
        /// <summary>
        /// Initializes a new instance of the Usage class.
        /// </summary>
        public Usage()
        {
          CustomInit();
        }

        /// <summary>
        /// Initializes a new instance of the Usage class.
        /// </summary>
        /// <param name="unit">Gets the unit of measurement. Possible values
        /// include: 'Count', 'Bytes', 'Seconds', 'Percent', 'CountsPerSecond',
        /// 'BytesPerSecond'</param>
        /// <param name="currentValue">Gets the current count of the allocated
        /// resources in the subscription.</param>
        /// <param name="limit">Gets the maximum count of the resources that
        /// can be allocated in the subscription.</param>
        /// <param name="name">Gets the name of the type of usage.</param>
        public Usage(UsageUnit unit, int currentValue, int limit, UsageName name)
        {
            this.Unit = unit;
            this.CurrentValue = currentValue;
            this.Limit = limit;
            this.Name = name;
            CustomInit();
        }

        /// <summary>
        /// An initialization method that performs custom operations like setting defaults
        /// </summary>
        partial void CustomInit();

        /// <summary>
        /// Gets the unit of measurement. Possible values include: 'Count',
        /// 'Bytes', 'Seconds', 'Percent', 'CountsPerSecond', 'BytesPerSecond'
        /// </summary>
        [Newtonsoft.Json.JsonProperty(PropertyName = "unit")]
        public UsageUnit Unit { get; set; }

        /// <summary>
        /// Gets the current count of the allocated resources in the
        /// subscription.
        /// </summary>
        [Newtonsoft.Json.JsonProperty(PropertyName = "currentValue")]
        public int CurrentValue { get; set; }

        /// <summary>
        /// Gets the maximum count of the resources that can be allocated in
        /// the subscription.
        /// </summary>
        [Newtonsoft.Json.JsonProperty(PropertyName = "limit")]
        public int Limit { get; set; }

        /// <summary>
        /// Gets the name of the type of usage.
        /// </summary>
        [Newtonsoft.Json.JsonProperty(PropertyName = "name")]
        public UsageName Name { get; set; }

        /// <summary>
        /// Validate the object.
        /// </summary>
        /// <exception cref="Microsoft.Rest.ValidationException">
        /// Thrown if validation fails
        /// </exception>
        public virtual void Validate()
        {
            if (this.Name == null)
            {
                throw new Microsoft.Rest.ValidationException(Microsoft.Rest.ValidationRules.CannotBeNull, "Name");
            }
        }
    }
}

File: Samples/azure-storage/Azure.NodeJS/models/usage.js
/*
 */

'use strict';

var models = require('./index');

/**
 * @class
 * Initializes a new instance of the Usage class.
 * @constructor
 * Describes Storage Resource Usage.
 *
 * @member {string} unit Gets the unit of measurement. Possible values include:
 * 'Count', 'Bytes', 'Seconds', 'Percent', 'CountsPerSecond', 'BytesPerSecond'
 *
 * @member {number} currentValue Gets the current count of the allocated
 * resources in the subscription.
 *
 * @member {number} limit Gets the maximum count of the resources that can be
 * allocated in the subscription.
 *
 * @member {object} name Gets the name of the type of usage.
 *
 * @member {string} [name.value] Gets a string describing the resource name.
 *
 * @member {string} [name.localizedValue] Gets a localized string describing
 * the resource name.
 *
 */
function Usage() {
}

/**
 * Defines the metadata of Usage
 *
 * @returns {object} metadata of Usage
 *
 */
Usage.prototype.mapper = function () {
  return {
    required: false,
    serializedName: 'Usage',
    type: {
      name: 'Composite',
      className: 'Usage',
      modelProperties: {
        unit: {
          required: true,
          serializedName: 'unit',
          type: {
            name: 'Enum',
            allowedValues: [ 'Count', 'Bytes', 'Seconds', 'Percent', 'CountsPerSecond', 'BytesPerSecond' ]
          }
        },
        currentValue: {
          required: true,
          serializedName: 'currentValue',
          type: {
            name: 'Number'
          }
        },
        limit: {
          required: true,
          serializedName: 'limit',
          type: {
            name: 'Number'
          }
        },
        name: {
          required: true,
          serializedName: 'name',
          type: {
            name: 'Composite',
            className: 'UsageName'
          }
        }
      }
    }
  };
};

module.exports = Usage;
